### PR TITLE
Add zammad module

### PR DESCRIPTION
**Pull request by adb-sh**
deploy zammad service

### CODE REVIEW

File: modules/management/zammad.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        user = username;
+        passwordFile = config.sops.secrets.zammad_db_pass.path;
+      };
+    };
+
+    nginx = {
+      enable = true;
+      recommendedProxySettings = true;
+      virtualHosts = {
+        "project.${config.deployment-dd-ix.domain}" = {
```
Comment (tanneberger) [inline modules/management/zammad.nix:44]: maybe a bit to generic: suggestions: `kanban.dd-ix.net`or `tickets.dd-ix.net`

File: modules/management/zammad.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+              proxyPass = "http://127.0.0.1:${toString config.services.zammad.port}";
+              proxyWebsockets = true;
+            };
+          };
+        };
+      };
+    };
+  };
+
+  networking.firewall.allowedTCPPorts = [ 80 443 ];
```
Comment (tanneberger) [inline modules/management/zammad.nix:58]: not necessary (and even if pls move it into some nginx.nix or smth like this)

File: modules/management/zammad.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      package = pkgs.zammad;
+      port = http_port;
+      websocketPort = ws_port;
+      database = {
+        createLocally = false;
+        type = "PostgreSQL";
+        host = "/run/postgresql";
+        name = database_name;
+        user = username;
+        passwordFile = config.sops.secrets.zammad_db_pass.path;
```
Comment (tanneberger) [inline modules/management/zammad.nix:36]: I dont see the point in the config where the database users password is set. 

(taken from vaultwarden)
```nix
systemd.services.vaultwarden-pgsetup = {
    description = "Prepare postgres database";
    wantedBy = [ "multi-user.target" ];
    after = [ "networking.target" "postgresql.service" ];
    serviceConfig.Type = "oneshot";

    path = [ pkgs.sudo config.services.postgresql.package ];
    script = ''
      # create postgres user with the specified password
      sudo -u ${config.services.postgresql.superUser} psql -c "ALTER ROLE vaultwarden WITH PASSWORD '$(cat ${config.sops.secrets.postgres_vaultwarden.path})'"
    '';
  };
```

alternatively, you also can put it into 

`systemd.services.postgresql.postStart`

**Review by tanneberger — CHANGES_REQUESTED** [review]

**Comment by tanneberger** [comment]
And I would also refer you to our ifsr config where zammad already runs in production under reasonable load and criticality.
https://github.com/fsr/fruitbasket/blob/main/modules/zammad.nix

File: modules/management/zammad.nix (hunk 1 above)
Comment (adb-sh) [reply modules/management/zammad.nix:44]: wanted to keep the `tickets`-Subdomain free for a potential Pretix-installation. what about `management` or `orga` ?

Comment (tanneberger) [reply modules/management/zammad.nix:30]: orga sounds good

**Review by tanneberger — APPROVED** [review]

**Pull request merged**